src/hypothesis/strategies.py

```diff
@@ -695,6 +695,32 @@ def text(
 
 @cacheable
 @defines_strategy
+def strings_matching_regex(regex):
+    """Return strategy that generates strings that match given regex.
+
+    Regex can be either a string or compiled regex (through
+    :func:`re.compile`).
```

Comment:
Add "Only unicode regexes are supported at this time.", and 

    :func:`python:re.compile`

again for cross-reference reliability.